students/EricAdams/trapezoidal_rule/decorator_trapezoidal_rule.py

```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env python
+
+import trapezoidal_rule as tr
+
+
+def area_under_function(func):
+    def trapz_area_under_curve(func, n, a, b, step):
+        '''
+        Calculate the area under a curve using the
+        trapezoidal rule.
+        :func = the function that defines the curve
```

Comment:
Is this a standard format for docstrings? I don' think I've seen it. Ideally you want to use a standardaized format so they can be nicely foramttted by doc tools.

I usually use the default sphinx format:

```
:param func: the function that defines the curve
:type func: a callable object, with the first argument as the varying axis
```
Bu there are others, like the "numpy" format:

https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#docstring-standard

And there may be others -- but better to use a common standard in any case.